### ISSUE TITLE
Button containing the aria-haspopup attribute identified as menu button submenu

### ISSUE THREAD

**Issue by Gabx-dev** [nvaccess/nvda#14159]
NVDA reports buttons containing the "aria-haspopup" attribute set to "dialog" as "menu button  subMenu". These properties don't reffflect the actual behavior and semantic meaning of the component, which in turn can potentially cause confusion or lead to mistakenly filled bugs in an accessibility testing scenario.

The component should be read as per [this documentation on the aria-haspopup attribute by MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup).

### Steps to reproduce:
1. Turn on Speech Viewer under NVDA > Tools > Speech Viewer.
2. Open the zipped HTML file attached in this issue with Google Chrome.
3. Navigate to the document with F6, if necessary.
4. Ensure NVDA is in browse mode with NVDA + space.
5. Navigate with up and down arrows, if necessary, so NVDA reads the button.
6. Note the button is read as "menu button  subMenu    Open di alog"
### Actual behavior:
menu button  subMenu    Open di alog
### Expected behavior:
popup button    Open di alog
### NVDA logs, crash dumps and other attachments:
[aria-haspopup-test.zip](https://github.com/nvaccess/nvda/files/9592013/aria-haspopup-test.zip)

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA version 2022.2.2
#### Windows version:
Windows 11 home Single Language version 21H2 (Operating System Build 22000.978)
#### Name and version of other software in use when reproducing the issue:
Google Chrome 105.0.5195.127 (Official version) 64 bits (cohort: Stable) 
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

**Comment by seanbudd** [comment]
This is a duplicate of #8235
